test/traversals/greedy_color.jl

```diff
@@ -0,0 +1,32 @@
+@testset "Greedy Coloring" begin
+  
+    g3 = StarGraph(10)
+
+    for g in testgraphs(g3)
+        for op1 in (true, false), op2 in (true, false)
+            C = @inferred(greedy_color(g, reps=50, sort_degree=op1, parallel=op2))
+            @test C.num_colors == 2
+        end
+    end
+    
+    g4 = PathGraph(20)
+    g5 = CompleteGraph(20)
+
+    for graph in [g4, g5]
+        for g in testgraphs(graph)
+            for op1 in (true, false), op2 in (true, false)
+                C = @inferred(greedy_color(g, reps=50, sort_degree=op1, parallel=op2))
+        
+                correct = true
+                C.num_colors > maximum(degree(g))+1 && (correct = false)
+                for e in edges(g)
+                    C.colors[Int(src(e))] == C.colors[Int(dst(e))] && (correct = false)
+                end
+
+                @test correct
```

Comment:
If this test fails, we won't know which component failed. Better to `@test` explicitly on lines 21 and 23.